src/marshaller/HipieDDL.js

```diff
@@ -1287,7 +1319,7 @@
         var datasourceRequestOptimizer = new DatasourceRequestOptimizer();
         for (var key in this.visualizationRequests) {
             var item = this.visualizationRequests[key];
-            if (item.updateVisualization.type !== "GRAPH") {
+            if (!this.skipClear && item.updateVisualization.type !== "GRAPH") {
```

Comment:
primeData never clears the visualization but later VisualizationRequestOptimizer() may clear.  Just checking that this is intended.